test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4247,45 +4250,71 @@ def _load_configuration(self, infile, objtype=None):
         """
         if os.path.isfile(infile):
             conf = {}
-            f = open(infile, 'r')
-            # load all objects from the Pickled file
-            while True:
-                try:
-                    conf = cPickle.load(f)
-                except:
-                    break
-            f.close()
-
-            if objtype and objtype in conf:
-                conf = conf[objtype]
-            else:
-                # load all object types that could be in infile
-                newconf = {}
-                for ky in [MGR_OBJ_SERVER, MGR_OBJ_SCHED, MGR_OBJ_NODE]:
-                    if ky not in conf:
-                        conf[ky] = {}
-                    newconf = dict(newconf.items() + conf[ky].items())
-                conf = newconf
-
+            with open(infile, 'r') as f:
+                conf = json.load(f)
             for k, v in conf.items():
                 fn = self.du.create_temp_file()
-                with open(fn, 'w') as fd:
-                    # handle server data saved as output of qmgr commands
-                    # by piping data back into qmgr
-                    if k.startswith('qmgr_'):
-                        qmgr = os.path.join(self.client_conf['PBS_EXEC'],
-                                            'bin', 'qmgr')
-                        fd.write("\n".join(v))
-                        self.du.run_cmd(self.hostname, [qmgr], cstdin=fd,
-                                        sudo=True)
-                    else:
-                        fd.write("\n".join(v))
-                        # append the last line
-                        fd.write("\n")
-                        self.du.run_cmd(self.hostname, ['cp', fn, k],
-                                        sudo=True)
+                qmgr = os.path.join(self.client_conf['PBS_EXEC'],
+                                    'bin', 'qmgr')
+                if k.startswith('qmgr_'):
+                    #with open(fn, 'w') as fd:
+                    #    fd.write("#!/bin/bash\n")
+                    #    fd.write("\n".join(v))
+                    fd, path = tempfile.mkstemp(prefix="runcmd_")
+                    os.chmod(path, 0o700)
+                    with open(path, 'a') as f:
+                        f.write('\n'.join(v))
+                    qmgr_cmd = qmgr +" < "+path
+                    d = self.du.run_cmd(self.hostname, qmgr_cmd, sudo=True, as_script=True)
+                    os.unlink(path)
+                    # Load site hooks
```

Comment:
As below else block is handling pbs conf loading, can you please modify the comment